File: su/src/main/java/org/sqlbuilder/sumo/KBLoader.java
package org.sqlbuilder.sumo;

public class KBLoader
{
	public static Kb kb;

	public static String getPath()
	{
		String kbPath = System.getProperty("sumopath");
		if (kbPath == null)
		{
			kbPath = System.getenv("SUMOHOME");
		}
		assert kbPath != null;
		return kbPath;
	}

	public static final String[] ALL_FILES = null;

	public static final String[] CORE_FILES = {"Merge.kif", "Mid-level-ontology.kif"};

	public static final String[] SAMPLE_FILES = {"Merge.kif", "Mid-level-ontology.kif", "Communications.kif"};

	public static String[] getScope()
	{
		String scope = System.getProperties().getProperty("SCOPE", "all");
		switch (scope)
		{
			case "all":
				return ALL_FILES;
			case "core":
				return CORE_FILES;
			case "samples":
				return SAMPLE_FILES;
			default:
				return scope.split("\\s");
		}
	}

	public static Kb loadKb()
	{
		return loadKb(getScope());
	}

	public static Kb loadKb(final String[] files)
	{
		String kbPath = getPath();
		Kb kb = new Kb(kbPath);
		System.out.printf("Kb building%n");
		boolean result = kb.make(files);
		assert result;
		System.out.printf("%nKb built%n");
		return kb;
	}

	public void load()
	{
		kb = loadKb();
	}
}

File: su/src/main/java/org/sqlbuilder/su/KBLoader.kt
package org.sqlbuilder.su

import org.sqlbuilder.common.Progress.resetGranularity
import org.sqlbuilder.common.Progress.setGranularity
import org.sqlbuilder.common.Progress.traceHeader
import org.sqlbuilder.common.Progress.traceTailer

open class KBLoader {

    open fun load() {
        kb = loadKb()
    }

    companion object {

        var kb: Kb? = null

        val path: String
            get() {
                var kbPath = System.getProperty("sumopath")
                if (kbPath == null) {
                    kbPath = System.getenv("SUMOHOME")
                }
                checkNotNull(kbPath)
                return kbPath
            }

        val ALL_FILES: Array<String>? = null

        val CORE_FILES: Array<String> = arrayOf("Merge.kif", "Mid-level-ontology.kif")

        val SAMPLE_FILES: Array<String> = arrayOf("Merge.kif", "Mid-level-ontology.kif", "Communications.kif")

        val scope: Array<String>?
            get() {
                val scope = System.getProperties().getProperty("SCOPE", "all")
                return when (scope) {
                    "all"     -> ALL_FILES
                    "core"    -> CORE_FILES
                    "samples" -> SAMPLE_FILES
                    else      -> scope.split("\\s".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
                }
            }

        fun loadKb(files: Array<String>? = scope): Kb {
            val kbPath: String = path
            val kb = Kb(kbPath)
            traceHeader("sumo", path)
            setGranularity(1L)
            val n = kb.make(files)
            resetGranularity()
            traceTailer(n.toLong())
            return kb
        }
    }
}
